main.go

```diff
@@ -455,3 +473,67 @@ var outputHeader = lipgloss.NewStyle().Foreground(lipgloss.Color("#F1F1F1")).Bac
 func printFilenameOutput(filename string) {
 	fmt.Println(lipgloss.JoinHorizontal(lipgloss.Center, outputHeader.String(), filename))
 }
+
+type Positions struct {
+	Left   float64
+	Center float64
+	Right  float64
+	Top    float64
+}
+
+func getPositions(config Config, x, y, imgW, controlsWidth float64) Positions {
+	return Positions{
+		Left:   config.Margin[left] + x + controlsWidth,
+		Center: imgW / 2,
+		Right:  imgW - (config.Margin[right] + x),
+		Top:    config.Margin[top] + y,
+	}
+}
+
+func (title Title) Validate(value string) error {
+	if title.Text == "" || title.Text == "-" {
+		return errors.New("Invalid title provided")
+	}
+	if title.Position != "left" && title.Position != "center" && title.Position != "right" {
+		return errors.New("Invalid title position. Must be one of \"left\", \"center\", or \"right\"")
+	}
+	return nil
+}
+
+// NewWindowTitle returns a title element with the given text.
+func NewWindowTitle(config Config, positions Positions, scale float64, s *chroma.Style) (*etree.Element, error) {
+	titleText := config.Title.Text
+	err := config.Title.Validate(titleText)
+	if err != nil {
+		return nil, err
+	}
+	if titleText == "auto" {
+		titleText = filepath.Base(config.Input)
+	}
+	x := 0.0
+	y := positions.Top
+	var anchor string
+	switch config.Title.Position {
```

Comment:
Left, center, .... they should be constant you define in the package, unexported ones

Reply:
@ccoVeille with this you mean for example, in `type Positions`:

```go
positions.Left{ x: config.Margin[left] + x + controlsWidth, anchor: "start" }
```

or how do you mean to do it?

Reply:
I just meant

```go
const (
     positionLeft       = "left"
     positionCenter = "center"
     positionRight.   = "right"
)
```